rust_src/src/editfns.rs

```diff
@@ -182,3 +182,43 @@ pub fn char_after(mut pos: LispObject) -> LispObject {
         }
     }
 }
+
+/// Return a copy of STRING with text properties added.
+/// First argument is the string to copy.
+/// Remaining arguments form a sequence of PROPERTY VALUE pairs for text
+/// properties to add to the result.
+/// usage: (propertize STRING &rest PROPERTIES)  */
+#[lisp_fn(min = "1")]
+pub fn propertize(args: &mut [LispObject]) -> LispObject {
+    /* Number of args must be odd. */
+    if args.len() & 1 == 0 {
+        error!("Wrong number of arguments");
+    }
+
+    let mut it = args.iter();
+
+    let first = it.next().unwrap(); // safe, there is at least 1 arg
+    if !first.is_string() {
```

Comment:
It should be fine to just do a `first.as_string_or_error()` here, and not use the result.

Reply:
Ok, I was trying to keep the C code's error.

Reply:
If I use `as_string_or_error` I can no long use the `as_raw` method in the call to Fcopy_sequence.

Reply:
Good point, I think it's fine to keep it the way it was then.

Reply:
I think the new version satisfies the expected style.